src/System.Runtime.Numerics/tests/BigInteger/cast_from.cs

```diff
@@ -549,11 +549,28 @@ public static void RunDoubleExplicitCastFromBigIntegerTests()
             bigInteger += BigInteger.One;
             VerifyDoubleExplicitCastFromBigInteger(Double.MaxValue, bigInteger);
 
+            // Double Explicit Cast from BigInteger: Double.MinValue - 1
+            bigInteger = new BigInteger(Double.MinValue);
+            bigInteger -= BigInteger.One;
+            VerifyDoubleExplicitCastFromBigInteger(Double.MinValue, bigInteger);
+
             // Double Explicit Cast from BigInteger: Random value > Double.MaxValue
             bigInteger = GenerateRandomBigIntegerGreaterThan(Double.MaxValue, s_random);
             bigInteger *= 2;
             VerifyDoubleExplicitCastFromBigInteger(Double.PositiveInfinity, bigInteger);
 
+            // Double Explicit Cast from BigInteger: Random value < -Double.MaxValue
+            VerifyDoubleExplicitCastFromBigInteger(Double.NegativeInfinity, -bigInteger);
+
+            // Double Explicit Cast from BigInteger: very large values (more than Int32.MaxValue bits) should be infinity
+            bigInteger = BigInteger.One;
+            for (int i = 0; i < 5; ++i)
```

Comment:
Did you mean to use the loop variable somewhere (in the shift)? I think it's unused right now.